shared/core/types/events.go

```diff
@@ -0,0 +1,13 @@
+package types
```

Comment:
Rename file to `fsm_events.go` so its not confused with `shared/messaging/events.go`